django/core/validators.py

```diff
@@ -459,7 +459,7 @@ class FileExtensionValidator:
     code = 'invalid_extension'
 
     def __init__(self, allowed_extensions=None, message=None, code=None):
-        self.allowed_extensions = allowed_extensions
+        self.allowed_extensions = sorted([allowed_extension.lower() for allowed_extension in allowed_extensions])
```

Comment:
Instead of using `lower()` this and the `lower()` on the extension in `__call__` could be replaced by `casefold()` since this is for Python >3.3.